From 68a1200f66e9008ca0a739b37b48c49453ca9d83 Mon Sep 17 00:00:00 2001
From: Earl Chew <earl_chew@yahoo.com>
Date: Fri, 10 May 2024 19:46:28 -0700
Subject: [PATCH] Restore empty string default for conf.env['icu-libs']

The reworked ICU libraries configuration code used [] as
default for conf.env['icu-libs']. This breaks dependency analysis
in samba_deps.py because SAMBA_SUBSYSTEM() expects deps to be
a string.

Signed-off-by: Earl Chew <earl_chew@yahoo.com>
Reviewed-by: Andrew Bartlett <abartlet@samba.org>
Reviewed-by: Douglas Bagnall <douglas.bagnall@catalyst.net.nz>
Reviewed-by: Volker Lendecke <vl@samba.org>

Autobuild-User(master): Andreas Schneider <asn@cryptomilk.org>
Autobuild-Date(master): Tue May 14 14:44:06 UTC 2024 on atb-devel-224
---
 lib/util/charset/wscript_configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/util/charset/wscript_configure b/lib/util/charset/wscript_configure
index c49b55a4fd4b..adae44eab5e4 100644
--- a/lib/util/charset/wscript_configure
+++ b/lib/util/charset/wscript_configure
@@ -46,3 +46,5 @@ if conf.CHECK_CFG(package='icu-i18n icu-uc',
     if not conf.CHECK_HEADERS('unicode/ustring.h', lib='icui18n'):
         conf.fatal('Found icui18n, but unicode/ustring.h is missing')
     conf.DEFINE('HAVE_UTF8_NORMALISATION', 1)
+else:
+    conf.env['icu-libs'] = ''
